pyxem/signals/diffraction_variance2d.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+"""Signal class for virtual diffraction contrast images.
+
+"""
+
+from hyperspy.signals import Signal2D
+
+from pyxem.signals.diffraction_variance1d import DiffractionVariance1D
+
+from pyxem.utils.expt_utils import radial_average
+
+import numpy as np
+
+
+class DiffractionVariance2D(Signal2D):
+    _signal_type = "diffraction_variance2d"
+
+    def __init__(self, *args, **kwargs):
+        Signal2D.__init__(self, *args, **kwargs)
+
+    def get_radial_profile(self, inplace=False, **kwargs):
+        """Return the radial profile of the diffraction variance signals.
+
+        Returns
+        -------
+        DiffractionVariance1D
+            radial_profile: :obj:`pyxem.signals.DiffractionVariance1D`
+            The radial profile of each diffraction variance pattern in the
+            DiffractionVariance2D signal.
+
+        See also
+        --------
+        :func:`pyxem.utils.expt_utils.radial_average`
+
+        Examples
+        --------
+        .. code-block:: python
+            profiles = ed.get_radial_profile()
+            profiles.plot()
+        """
+        radial_profiles = self.map(radial_average,
+                                   inplace=inplace, **kwargs)
+
+        radial_profiles.axes_manager.signal_axes[0].offset = 0
+        signal_axis = radial_profiles.axes_manager.signal_axes[0]
+
+        rp = DiffractionVariance1D(radial_profiles.as_signal1D(signal_axis))
+        rp_axis = rp.axes_manager.signal_axes[0]
```

Comment:
transfer axes function? (either use one, or write one for this (2D ---> 1D case)

Reply:
Current 2D->1D don't have it, so it shouldn't be part of this pull request I think.

Reply:
Let's write one for the 2D --> 1D case and put it in here. It perhaps wasn't so relevant before because we didn't do it in so many places in the code, but now we do it in more places we should have one. So it's in scope here I think.